docker/Dockerfile.dev

```diff
@@ -52,7 +52,7 @@ ARG CUDA_VERSION
 RUN if [ "$MACHINE_TYPE" = "gpu" ]; then export USE_CUDA=1; fi \
     && git clone https://github.com/pytorch/serve.git \
     && cd serve \
-    && git checkout ${BRANCH_NAME} \
+    && git checkout --track origin/release_0.4.0 \
```

Comment:
Why is the branch being hardcoded? It was parameter earlier

Reply:
This is a dev-change that should've been reverted in the cleanup, fixed.